File: src/SearchInsertPosition/searchInsertPosition.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    if(nums[0] < target && nums.length == 1) return 1;
    if(nums[0] >= target) return 0;
    for(var i = 1; i < nums.length; i++) {
        if(nums[i] >= target && target > nums[i - 1]) {
            return i;
        } 
    }
    return nums.length;
};

File: src/SearchInsertPosition/searchInsertPosition.java
public class Solution {
    public int searchInsert(int[] nums, int target) {
        int low = 0;int high = nums.length - 1;
        int middle = (low + high) / 2;
        while(low <= high) {
            middle = (low + high) / 2;
            if (nums[middle] < target) {
                low = middle + 1;
            } else if (nums[middle] > target) {
                high = middle - 1;
            } else {
                return middle;
            }
        }
        return low;
    }
}